### PR TITLE
Implement `Capability{Reader,Keeper}` traits

### PR DESCRIPTION
**Pull request by hu55a1n1**
Closes: #1769 

- [x] Add `CapabilityReader/Keeper` traits
- [x] Fix isolated crate build
- [x] Remove `ics26::dispatch` ctx clone. 
- [x] Derive `Display` for `ics24::Path` variant types. 
______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [x] Reviewed `Files changed` in the GitHub PR explorer.
- [x] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: modules/src/core/ics05_port/context.rs [hunk 1]
```diff
@@ -1,11 +1,68 @@
...
+    /// Check if the specified port_id is already bounded
+    fn is_bound(&self, port_id: PortId) -> bool {
+        self.get_capability(format!("{}", PortsPath(port_id)))
+            .is_ok()
+    }
+
+    /// Authenticate a capability key against a port_id by checking if the capability was previously
+    /// generated and bound to the specified port
+    fn authenticate(&self, port_id: PortId, capability: &Capability) -> bool {
+        self.authenticate_capability(format!("{}", PortsPath(port_id)), capability)
```
Comment (romac) [inline modules/src/core/ics05_port/context.rs:28]: Can we just do `to_string()` here instead?

```suggestion
        self.authenticate_capability(PortsPath(port_id).to_string(), capability)
```

File: modules/src/core/ics05_port/context.rs [hunk 2]
```diff
@@ -1,11 +1,68 @@
...
+            self.new_capability(format!("{}", PortsPath(port_id)))
+        }
+    }
 }
 
-//  Result<Capability, Error>//return Ok(Capability::new());
+pub trait CapabilityKeeper {
+    /// Create a new capability with the given name.
+    /// Return an error if the capability was already taken.
+    fn new_capability(&mut self, name: impl AsRef<str>) -> Result<Capability, Error>;
```
Comment (romac) [inline modules/src/core/ics05_port/context.rs:47]: Is `AsRef<str>` really the best representation we can find for capabilities names? What are the operations on the capability name that are actually required? Perhaps we can define a trait for that and use `impl CapabilityName` instead?

File: modules/src/core/ics05_port/error.rs [hunk 3]
```diff
@@ -8,6 +8,10 @@ define_error! {
             { port_id: PortId }
             | e | { format_args!("Port {0} is unknown", e.port_id) },
```
Comment (romac) [inline modules/src/core/ics05_port/error.rs:9]:
```suggestion
            | e | { format_args!("port '{0}' is unknown", e.port_id) },
```

File: modules/src/core/ics05_port/error.rs [hunk 4]
```diff
@@ -8,6 +8,10 @@ define_error! {
...
 
+        PortAlreadyBound
+            { port_id: PortId }
+            | e | { format_args!("Port {0} is already bound", e.port_id) },
```
Comment (romac) [inline modules/src/core/ics05_port/error.rs:13]:
```suggestion
            | e | { format_args!("port '{0}' is already bound", e.port_id) },
```

**Review by romac — CHANGES_REQUESTED** [review]

File: modules/src/core/ics05_port/context.rs (hunk 2 above)
Comment (hu55a1n1) [reply modules/src/core/ics05_port/context.rs]: I just looked up the SDK impl and it seems capability names are expected to be non-empty strings (i.e. `strings.TrimSpace(name) != ""`) -> https://github.com/cosmos/cosmos-sdk/blob/master/x/capability/keeper/keeper.go#L230 , so I think it is better to use a newtype.

Comment (romac) [reply modules/src/core/ics05_port/context.rs]: Sweet!

**Review by romac — APPROVED** [review]
Awesome work! Just needs a merge from `master` and we're good to go :)

**Pull request merged**